src/main/java/com/google/devtools/build/lib/analysis/RuleContext.java

```diff
@@ -2091,6 +2091,11 @@ public boolean isVisible(TransitiveInfoCollection prerequisite) {
       return RuleContext.isVisible(target.getAssociatedRule(), prerequisite);
     }
 
+    @Nullable
+    Aspect getMainAspect() {
```

Comment:
Readability: Can you reuse the same implementation as `getMainAspect` in the RuleContext? That would be still some duplication, but at least not a different one.